lib/utils/emitDeprecationWarning.mjs

```diff
@@ -0,0 +1,25 @@
+import process from 'node:process';
+
+import { assertString } from './validateTypes.mjs';
+
+// Each code must be unique.
+const CODES = {
+	COMMONJS_PLUGINS: '001',
+	COMMONJS_NODEJS_API: '002',
+	RESULT_OUTPUT_PROPERTY: '003',
+};
+
+/**
+ * Use this function for deprecation warnings, instead of `process.emitWarning()`.
+ *
+ * @param {string} message
+ * @param {keyof CODES} codeKey
+ * @param {string} detail
+ * @returns {void}
+ */
+export default function emitDeprecationWarning(message, codeKey, detail) {
+	const code = CODES[codeKey];
+
+	assertString(code);
+	process.emitWarning(message, { type: 'DeprecationWarning', code: `stylelint:${code}`, detail });
```

Comment:
question

Why wouldn't the details not be mapped to the code keys like the codes are?
Are you expecting using the same code with different details?

Reply:
I want to put a `message` and `detail` in the same place because they are strongly related.

`detail` has not been repeated yet, so I don't think it's worth coupling the code and detail for now.